--- foss-2020a.orig/installer/site-config/Makefile.default	2021-01-12 14:17:52.000000000 +0800
+++ foss-2020a/installer/site-config/Makefile.default	2021-01-19 11:03:38.750000000 +0800
@@ -41,6 +41,12 @@
 # This string is a {TCL list of words}.
 # Include \n to have pretty linebreaks in the modulefile.
 MODULEFILE_DEPENDENCIES={\n\
+    foss         foss/2020a \n\
+    Python       Python/3.8.2-GCCcore-9.3.0 \n\
+	Java         Java/1.8 \n\
+	ant 		 ant/1.10.1-Java-1.8 \n\
+    MDSplus      MDSplus/7.96.12-GCCcore-9.3.0  \n\
+    Blitz++      Blitz++/1.0.2 \n\
 }
 
 # Test for this jar file in the CLASSPATH:
@@ -50,18 +56,18 @@
 BLITZPCFILE=blitz.pc
 
 # Test that these module names are loaded for UAL compilation (compile time):
-UAL_MODULES =
+UAL_MODULES = Blitz++
 # Additionally test for these module names in case of IMAS_PGI=yes IMAS_IFORT=yes,.. see below.
 UAL_MODULES_CPP =
-UAL_MODULES_MDSPLUS =
+UAL_MODULES_MDSPLUS = MDSplus
 UAL_MODULES_UDA =
 UAL_MODULES_G95 =
 UAL_MODULES_NAGFOR =
 UAL_MODULES_PGI =
 UAL_MODULES_INTEL =
-UAL_MODULES_JAVA =
+UAL_MODULES_JAVA = Java
 UAL_MODULES_PYTHON2 =
-UAL_MODULES_PYTHON3 =
+UAL_MODULES_PYTHON3 = Python
 
 # Additional search path to local IMAS scripts that is added to $PATH by IMAS module
 IMAS_EXTRA_PATH = $(IMAS_HOME)/extra/bin
