# S.D. Pinches, ITER Organization, 20200403
# Patch to correctly install PSPLINE v2.0.0
diff -Nru pspline-2.0.0-orig/Makefile pspline-2.0.0/Makefile
--- pspline-2.0.0-orig/Makefile	2019-12-23 17:17:15.000000000 +0100
+++ pspline-2.0.0/Makefile	2020-04-03 13:36:57.747133891 +0200
@@ -69,7 +69,7 @@
 	@cp $(PSPDIR)/lookup_test.output $(TESTDR)
 	@cp $(EZSDIR)/ezspline_test.ref $(TESTDR)
 	@cp $(EZSDIR)/qk_pspline.output $(TESTDR)
-	@cp $(CZSDIR)/czspline_test.out $(TESTDR)
+	@cp $(CZSDIR)/czspline_test.output $(TESTDR)
 	@test ! -f $(EXELUP) || cp $(EXELUP) $(TESTDR)
 	@test ! -f $(EXEPSP) || cp $(EXEPSP) $(TESTDR)
 	@test ! -f $(EXEQKS) || cp $(EXEQKS) $(TESTDR)